extension/content/components/recipes/details/DetailsHeader.js

```diff
@@ -26,6 +26,13 @@ export default function DetailsHeader() {
     history.push(`/${environmentKey}/recipes/${recipeId}/edit`);
   };
 
+  const handleCopyClick = () => {
+    history.push({
+      pathname: `/${environmentKey}/recipes/new`,
+      state: { data },
```

Comment:
I actually think this should live at `/${environmentKey}/recipes/${recipeId}/clone` and then it can use the data right from the context.